api/tests/test_application_status_api.py

```diff
@@ -0,0 +1,243 @@
+
+    def test_unit_handler_can_create_validated_status(self):
+        response = self.unit_handler_client.post(
+            reverse("application_event_status-list"),
+            data={
+                "status": ApplicationEventStatus.VALIDATED,
+                "application_event_id": self.application_event.id,
+            },
+        )
+        assert_that(response.status_code).is_equal_to(201)
+        assert_that(ApplicationEventStatus.objects.latest("id").status).is_equal_to(
+            ApplicationEventStatus.VALIDATED
+        )
+
+    def test_service_sector_manager_can_create_approved_status(self):
+        response = self.manager_api_client.post(
+            reverse("application_event_status-list"),
+            data={
+                "status": ApplicationEventStatus.APPROVED,
+                "application_event_id": self.application_event.id,
+            },
+        )
+        assert_that(response.status_code).is_equal_to(201)
+        assert_that(ApplicationEventStatus.objects.latest("id").status).is_equal_to(
+            ApplicationEventStatus.APPROVED
+        )
+
+
+@pytest.mark.django_db
+class ReservationCreationOnStatusCreationTestCase(ApplicationStatusBaseTestCase):
+    def test_creating_approved_status_creates_reservations(self):
+        response = self.manager_api_client.post(
+            reverse("application_event_status-list"),
+            data={
+                "status": ApplicationEventStatus.APPROVED,
+                "application_event_id": self.application_event.id,
+            },
+        )
+        assert_that(response.status_code).is_equal_to(201)
+        assert_that(RecurringReservation.objects.count()).is_equal_to(1)
+        assert_that(Reservation.objects.count()).is_greater_than(0)
+
+    def test_correct_amount_of_reservations_are_created_when_weekly(self):
+        response = self.manager_api_client.post(
+            reverse("application_event_status-list"),
+            data={
+                "status": ApplicationEventStatus.APPROVED,
+                "application_event_id": self.application_event.id,
+            },
+        )
+        assert_that(response.status_code).is_equal_to(201)
+        assert_that(RecurringReservation.objects.count()).is_equal_to(1)
+        assert_that(Reservation.objects.count()).is_equal_to(4)
+
+    def test_correct_amount_of_reservations_are_created_when_bi_weekly(self):
+        self.application_event.biweekly = True
+        self.application_event.save()
+        response = self.manager_api_client.post(
+            reverse("application_event_status-list"),
+            data={
+                "status": ApplicationEventStatus.APPROVED,
+                "application_event_id": self.application_event.id,
+            },
+        )
+        assert_that(response.status_code).is_equal_to(201)
+        assert_that(RecurringReservation.objects.count()).is_equal_to(1)
+        assert_that(Reservation.objects.count()).is_equal_to(2)
```

Comment:
Could be good to add test for the date thingy.

Reply:
added the test